src/lib.rs

```diff
@@ -2753,6 +2753,155 @@ pub fn is_maximal_matching(
     })
 }
 
+fn _graph_triangles(graph: &graph::PyGraph, node: usize) -> (usize, usize) {
+    let mut triangles: usize = 0;
+
+    let index = NodeIndex::new(node);
+    let mut neighbors: HashSet<NodeIndex> =
+        graph.graph.neighbors(index).collect();
+    neighbors.remove(&index);
+
+    for nodev in &neighbors {
+        triangles += graph
+            .graph
+            .neighbors(*nodev)
+            .filter(|&x| (x != *nodev) && neighbors.contains(&x))
+            .count();
+    }
+
+    let d: usize = neighbors.len();
+    let triples: usize = match d {
+        0 => 0,
+        _ => (d * (d - 1)) / 2,
+    };
+
+    (triangles / 2, triples)
+}
+
+/// Compute the transitivity of an undirected graph.
+///
+/// The transitivity of a graph is defined as:
+///
+/// .. math::
+///     `c=3 \times \frac{\text{number of triangles}}{\text{number of connected triples}}`
+///
+/// A “connected triple” means a single vertex with
+/// edges running to an unordered pair of others.
+///
+/// :param PyGraph graph: Graph to be used.
+///
+/// :returns: Transitivity.
+/// :rtype: float
+#[pyfunction]
+#[text_signature = "(graph, /)"]
+fn graph_transitivity(graph: &graph::PyGraph) -> f64 {
+    let (triangles, triples) =
+        graph
+            .graph
+            .node_indices()
```

Comment:
One thought I just had here is there any advantage to doing this iteration in parallel? It doesn't look like there is any dependence on order (as it's all addition) and there is no interaction with python so we don't need to worry about the gil. To make this multithreaded we can just create a Vec and use rayon's parallel iterator here, so it would be something like:

```suggestion
    let node_indices: Vec<NodeIndex> = graph.graph.node_indices();
    let (triangles, triples) = node_indices
            .par_iter()
```

I'm not sure if there will be a benefit to running this in parallel or not, but my feeling is that it would be faster (although I'm often wrong) so it's probably worth benchmarking to checking.

Reply:
Good suggestion. I did some benchmarks locally and it's indeed faster